src/parser.ts

```diff
@@ -348,6 +350,35 @@ export function generateParser(options: { language: string }): (input: string) =
 		['NormalChar']
 	);
 
+	// BackReference (XPath only)
+	const backReference: Parser<never> = map(
+		preceded(
+			BACKSLASH,
+			then(
+				map(
+					filter(codepoint, charRangePredicate(asCodepoint('1'), asCodepoint('9')), [
+						'digit'
+					]),
+					codepoint => codepoint - ZERO_CODE_POINT
+				),
+				star(
+					map(
+						filter(codepoint, charRangePredicate(ZERO_CODE_POINT, asCodepoint('9')), [
+							'digit'
+						]),
+						codepoint => codepoint - ZERO_CODE_POINT
+					)
+				),
+				(firstDigit, restDigits) => {
+					restDigits.reduce((sum, digit) => sum * 10 + digit, firstDigit);
```

Comment:
Creative! As an alternative, you could use the `recognize` combinator (and isDigit from below), with something like:
```
const backReferenceNumber: Parser<number> = map(
    recognize(
        preceded(
            filter(codepoint, charRangePredicate(asCodepoint('1'), asCodepoint('9')), ['digit']),
            star(filter(codepoint, isDigit, ['digit']))
        )
    ),
    str => parseInt(str, 10)
);
```